File: flutter-exercises/meals_app/lib/widgets/meal_item.dart
import 'package:flutter/material.dart';
import 'package:meals_app/models/meal.dart';
import 'package:meals_app/widgets/meal_item_trait.dart';
import 'package:transparent_image/transparent_image.dart';

class MealItem extends StatelessWidget {
  const MealItem({
    super.key,
    required this.meal,
    required this.onSelectMeal,
  });

  final Meal meal;
  final void Function(Meal meal) onSelectMeal;

  String get complexityText {
    return meal.complexity.name[0].toUpperCase() + meal.complexity.name.substring(1).toLowerCase();
  }

    String get affordabilityText {
    return meal.affordability.name[0].toUpperCase() + meal.affordability.name.substring(1).toLowerCase();
  }

  @override
  Widget build(BuildContext context) {
    return Card(
      margin: const EdgeInsets.symmetric(vertical: 10, horizontal: 4),
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10),),
      clipBehavior: Clip.hardEdge,
      elevation: 2,
      child: InkWell(
        onTap: () { onSelectMeal(meal); },
        child: Stack(   // shows items on top of each other
          children: [
            Hero( // Hero animates widgets between screens
              tag: meal.id,
              child: FadeInImage(
                placeholder: MemoryImage(kTransparentImage), 
                image: NetworkImage(meal.imageUrl),
                fit: BoxFit.cover,
                height: 200,
                width: double.infinity,
                ),
            ),
              Positioned(
                bottom: 0,    // positioned on the bottom of the above fadeinimage
                left: 0,      // stretching from left to right
                right: 0,
                child: Container(
                  color: Colors.black54,
                  padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 20),
                  child: Column(
                    children: [
                      Text(
                        meal.title,
                        maxLines: 2,
                        textAlign: TextAlign.center,
                        softWrap: true,
                        overflow: TextOverflow.ellipsis,  // ...
                        style: const TextStyle(
                          fontSize: 20,
                          fontWeight: FontWeight.bold,
                          color: Colors.white,
                        ),
                      ),
                      const SizedBox(height: 10,),
                      Row(
                        mainAxisAlignment: MainAxisAlignment.center,
                        children: [
                          MealItemTrait(
                            icon: Icons.schedule, 
                            label: '${meal.duration} min'
                            ),
                            const SizedBox(width: 10),
                            MealItemTrait(icon: Icons.trending_up_rounded, label: complexityText),
                            const SizedBox(width: 10),
                            MealItemTrait(icon: Icons.attach_money_sharp, label: affordabilityText)
                        ],
                      ),
                    ],
                  ),
              ),),
          ],
        ),
      ),
    );
  }
}

File: reactjs-exercises/meals_app/components/MealItem.js
import { View, Text, Pressable, StyleSheet, Image } from "react-native"

function MealItem ({mealObj}) {
    return <View style={styles.mealItem}>
        <Pressable android_ripple={{color: '#ccc'}}>
            <View>
                <Image source={{uri: mealObj.imageUrl}} style={styles.image}/>
                <Text style={styles.title}>{mealObj.title}</Text>
            </View>
            <View style={styles.infoBox}>
                <Text style={styles.infoItem}>{mealObj.duration}m</Text>
                <Text style={styles.infoItem}>{mealObj.complexity}</Text>
                <Text style={styles.infoItem}>{mealObj.affordability}</Text>
            </View>
        </Pressable>
    </View>
}

export default MealItem;

const styles = StyleSheet.create({
    image: {
        width: '100%',
        height: 200
    },
    title: {
        fontWeight: 'bold',
        textaAlign: 'center',
        fontSize: 18,
        padding: 10
    },
    mealItem: {
        margin: 10,
        borderRadius: 8,
        overflow: 'hidden',
        backgroundColor: '#FAFAFA',
        elevation: 4
    },
    infoBox: {
        margin: 10,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center'
    },
    infoItem: {
        marginHorizontal: 6,
        fontSize: 12
    }
});